query/src/main/java/io/keen/client/java/KeenQueryClient.java

```diff
@@ -0,0 +1,704 @@
+         */
+        protected HttpHandler getDefaultHttpHandler() throws Exception {
+            return new UrlConnectionHttpHandler();
+        }
+
+        /**
+         * Gets the {@link HttpHandler} that this builder is currently configured to use for making
+         * HTTP requests. If null, a default will be used instead.
+         *
+         * @return The {@link HttpHandler} to use.
+         */
+        public HttpHandler getHttpHandler() {
+            return httpHandler;
+        }
+
+        /**
+         * Sets the {@link HttpHandler} to use for making HTTP requests.
+         *
+         * @param httpHandler The {@link HttpHandler} to use.
+         */
+        public void setHttpHandler(HttpHandler httpHandler) {
+            this.httpHandler = httpHandler;
+        }
+
+        /**
+         * Sets the {@link HttpHandler} to use for making HTTP requests.
+         *
+         * @param httpHandler The {@link HttpHandler} to use.
+         * @return This instance (for method chaining).
+         */
+        public QueryBuilder withHttpHandler(HttpHandler httpHandler) {
+            setHttpHandler(httpHandler);
+            return this;
+        }
+
+        /**
+         * Gets the default {@link KeenJsonHandler} to use if none is explicitly set for this builder.
+         * <p/>
+         * Subclasses must override this to provide a default {@link KeenJsonHandler}.
+         *
+         * @return The default {@link KeenJsonHandler}.
+         * @throws Exception If there is an error creating the {@link KeenJsonHandler}.
+         */
+        protected KeenJsonHandler getDefaultJsonHandler() throws Exception {
+            return new JacksonJsonHandler();
+        }
+
+        /**
+         * Gets the {@link KeenJsonHandler} that this builder is currently configured to use for
+         * handling JSON operations. If null, a default will be used instead.
+         *
+         * @return The {@link KeenJsonHandler} to use.
+         */
+        public KeenJsonHandler getJsonHandler() {
+            return jsonHandler;
+        }
+
+        /**
+         * Sets the {@link KeenJsonHandler} to use for handling JSON operations.
+         *
+         * @param jsonHandler The {@link KeenJsonHandler} to use.
+         */
+        public void setJsonHandler(KeenJsonHandler jsonHandler) {
+            this.jsonHandler = jsonHandler;
+        }
+
+        /**
+         * Sets the {@link KeenJsonHandler} to use for handling JSON operations.
+         *
+         * @param jsonHandler The {@link KeenJsonHandler} to use.
+         * @return This instance (for method chaining).
+         */
+        public QueryBuilder withJsonHandler(KeenJsonHandler jsonHandler) {
+            setJsonHandler(jsonHandler);
+            return this;
+        }
+
+        /**
+         * Gets the base URL to use for queries.
+         *
+         * @return The base URL to use.
+         */
+        public String getBaseURL() {return this.baseUrl;}
```

Comment:
Nit: According to our style rules there should be a space after the opening `{` and before the closing `}`. In fact technically this should be a full multi-line block but that's a fuzzier line, and I'm fine leaving it inline as long as the spacing's fixed.